### PR TITLE
removed validation from init

### PR DESCRIPTION
**Pull request by satya-k8s**

### CODE REVIEW

File: workflow-controller/controller-service/src/main/java/io/hyscale/controller/builder/K8sAuthConfigBuilder.java [hunk 1]
```diff
@@ -34,14 +42,33 @@
...
     /**
      * Gets {@link K8sConfigFileAuth} from {@link ControllerConfig} default config
      * @return {@link K8sConfigFileAuth}
      */
-    public AuthConfig getAuthConfig() {
+    public AuthConfig getAuthConfig() throws HyscaleException{
+    	validate(controllerConfig.getDefaultKubeConf(), true, ControllerErrorCodes.KUBE_CONFIG_NOT_FOUND);
         K8sConfigFileAuth k8sAuth = new K8sConfigFileAuth();
         k8sAuth.setK8sConfigFile(new File(controllerConfig.getDefaultKubeConf()));
         return k8sAuth;
```
Comment (SameeraGrandhi) [inline workflow-controller/controller-service/src/main/java/io/hyscale/controller/builder/K8sAuthConfigBuilder.java]: getAuthConfig(controllerConfig.getDefaultKubeConf()))
this works right ?

File: workflow-controller/controller-service/src/main/java/io/hyscale/controller/builder/K8sAuthConfigBuilder.java [hunk 2]
```diff
@@ -34,14 +42,33 @@
...
+    
+    public AuthConfig getAuthConfig(String kubeConfigPath) throws HyscaleException{
+    	validate(kubeConfigPath, true, ControllerErrorCodes.KUBE_CONFIG_NOT_FOUND);
+        K8sConfigFileAuth k8sAuth = new K8sConfigFileAuth();
+        k8sAuth.setK8sConfigFile(new File(kubeConfigPath));
+        return k8sAuth;
+    }
+    
+    private void validate(String path, boolean kubeConf, HyscaleErrorCode hyscaleErrorCode) throws HyscaleException {
+        File conffile = new File(path);
```
Comment (SameeraGrandhi) [inline workflow-controller/controller-service/src/main/java/io/hyscale/controller/builder/K8sAuthConfigBuilder.java]: camelcase , confFile

Comment (SameeraGrandhi) [reply workflow-controller/controller-service/src/main/java/io/hyscale/controller/builder/K8sAuthConfigBuilder.java]: It is a private method, no question of having docker config json . So why boolean and error code as a parameter ?

File: workflow-controller/controller-service/src/main/java/io/hyscale/controller/manager/impl/LocalRegistryManagerImpl.java [hunk 3]
```diff
@@ -76,32 +84,49 @@ public void init() throws HyscaleException {
...
+	public ImageRegistry getImageRegistry(String registry) throws HyscaleException {
+		ObjectMapper mapper = ObjectMapperFactory.jsonMapper();
+		try {
+			TypeReference<DockerConfig> dockerConfigTypeReference = new TypeReference<DockerConfig>() {
+			};
+			dockerConfig = mapper.readValue(new File(controllerConfig.getDefaultRegistryConf()),
+					dockerConfigTypeReference);
+		} catch (IOException e) {
+
+		}
```
Comment (SameeraGrandhi) [inline workflow-controller/controller-service/src/main/java/io/hyscale/controller/manager/impl/LocalRegistryManagerImpl.java]: Exception case ?

File: workflow-controller/controller-service/src/main/java/io/hyscale/controller/manager/impl/LocalRegistryManagerImpl.java [hunk 4]
```diff
@@ -132,5 +157,16 @@ private static String getHelperFunction(String registry) {
         String withHttpsAndSuffix = "https://" + registry + "/";
         return Arrays.asList(exactMatch, withHttps, withSuffix, withHttpsAndSuffix);
     }
-
+    
+
+    
+    private void validate(String path, boolean kubeConf, HyscaleErrorCode hyscaleErrorCode) throws HyscaleException {
+        File conffile = new File(path);
```
Comment (SameeraGrandhi) [inline workflow-controller/controller-service/src/main/java/io/hyscale/controller/manager/impl/LocalRegistryManagerImpl.java]: 1.camelcase confFile
2. No kubeconf here, remove it as a parameter
3. Cannot find docker config json as error code , no need of parameter

File: workflow-controller/controller-service/src/main/java/io/hyscale/controller/builder/K8sAuthConfigBuilder.java [hunk 5]
```diff
@@ -48,26 +47,26 @@
...
         k8sAuth.setK8sConfigFile(new File(kubeConfigPath));
         return k8sAuth;
     }
     
-    private void validate(String path, boolean kubeConf, HyscaleErrorCode hyscaleErrorCode) throws HyscaleException {
-        File conffile = new File(path);
-        if (conffile != null && !conffile.exists()) {
-            String confpath = kubeConf ? SetupConfig.getMountPathOfKubeConf(path) : SetupConfig.getMountOfDockerConf(path);
+    private void validate(String path) throws HyscaleException {
+        File confFile = new File(path);
```
Comment (tushar-tushar) [inline workflow-controller/controller-service/src/main/java/io/hyscale/controller/builder/K8sAuthConfigBuilder.java]: Add a null check, as this throws NPE

**Pull request merged**